internal/service/cluster/db/repo/repository.go

```diff
@@ -190,3 +190,15 @@ func (r *ClusterRepository) FindStaleAlarmDictionaries(ctx context.Context, data
 	e := psql.Quote("data_source_id").EQ(psql.Arg(dataSourceID)).And(psql.Quote("generation_id").LT(psql.Arg(generationID)))
 	return utils.Search[commonmodels.AlarmDictionary](ctx, r.Db, e)
 }
+
+// SetNodeClusterID sets the nodeClusterID value on cluster resources that may have arrived out of order
+func (r *ClusterRepository) SetNodeClusterID(ctx context.Context, nodeClusterName string, nodeClusterID uuid.UUID) (int, error) {
+	updates := models.ClusterResource{NodeClusterID: &nodeClusterID}
+	e := psql.Quote("node_cluster_name").EQ(psql.Arg(nodeClusterName)).And(psql.Quote("node_cluster_id").IsNull())
```

Comment:
should we be using the internal stuff to extract out the DB fields e.g utils.GetAllDBTagsFromStruct? Though i noticed in common we are also doing GetColumnsAndValues. In general not sure tbh...like doing reflections a few times just get the tags 🤔 (though i dont think performance is an issue here...but rather if idiomatic)

Reply:
We use the utilities to get the names when we need the full list, but when we need 1 or 2 (other than the primary key) we specify them by name.